cql3/expr/expression.cc

```diff
@@ -1779,13 +1779,35 @@ cql3::raw_value evaluate(const binary_operator& binop, const evaluation_inputs&
     return cql3::raw_value::make_value(boolean_type->decompose(binop_result));
 }
 
+cql3::raw_value evaluate(const conjunction& conj, const evaluation_inputs& inputs) {
+    for (const expression& child : conj.children) {
+        cql3::raw_value child_val = evaluate(child, inputs);
+        if (child_val.is_null()) {
+            // We could treat NULL as false, but this sounds wrong.
+            // The user can always just write `something IS NOT NULL`.
+            throw exceptions::invalid_request_exception("NULL value found inside AND conjunction");
```

Comment:
Why? It's very common.

```sql
WHERE a = 3 AND b = 4
```

if `a` is NULL, then a=3 is NULL, and then the conjunction is also NULL. So NULL isn't false, but it's not illegal either.

NULL converts everything it touches to NULL, except for IS NULL / IS NOT NULL. Like NaN in floating point.